### ISSUE TITLE
Support new version of Gecko Driver

### ISSUE THREAD

**Issue by tormozz48** [admc/wd#495]
Hi.

We have some new versions of Mozilla Firefox browsers which uses new GeckoDriver 0.15-0.19
Firefox of version 53.0 and above uses this driver which specification and development status described here: https://developer.mozilla.org/en-US/docs/Mozilla/QA/Marionette/WebDriver/status.

There some major changes and number of incompatibilities between old and new API versions.

Have you any plans and roadmaps for supporting this new API?

**Comment by vania-pooh** [comment]
As you probably know this updated spec has different JSON format. E.g. `sessionId` field as well as most of other data is now under `value` key. This is currently not supported by this client.

**Comment by betalb** [comment]
Does this means that it is not supported, and won't be supported by design or just no resources to support it?

**Comment by joeyparrish** [comment]
IE's latest driver seems to use the new protocol as well.  I have hacked some local changes together which seem to work with both protocol versions.  I had to make minor changes in both callbacks.js (return value formatting) and webdriver.js (finding session ID).  I only use WebDriver through Karma to send a URL to the remote browser, so it's possible that other changes would be necessary for a complete protocol update.

I will send a PR as soon as I can put one together, but again, it may be incomplete.
